src/openclsim/critical_path/simulation_graph.py

```diff
@@ -64,36 +150,286 @@ def __check_dep(self, dependency_list):
         ----------
         dependency_list : list
         """
-        # ..
+        assert isinstance(dependency_list, list), "list_dependencies must be a list"
+
+        dependency_list = [
+            (f"{self.__NODE_END_PREFIX} {end}", f"{self.__NODE_START_PREFIX} {start}")
+            for end, start in dependency_list
+        ]
         return dependency_list
 
     def __construct_graph(self):
         """
         Construct and return simulation graph (self.simulation_graph),
         the networkx diGraph representing the OpenClSim simulation,
-        so including all recorded activities and their dependencies (in time)
+        so including all recorded activities and their dependencies (in time).
 
         Returns
         -------
         simulation_graph : nx.DiGraph
             the networkx diGraph representing the OpenClSim simulation
         """
-        return nx.DiGraph()
+        self.simulation_graph = nx.DiGraph()
 
-    def __find_critical_edges(self):
+        self.__create_activity_edges()
+        self.__link_activity_edges()
+
+        # check if the resulting graph is acyclic
+        assert nx.is_directed_acyclic_graph(
+            self.simulation_graph
+        ), "The resulting graph appears to be cyclic!"
+
+        return self.simulation_graph
+
+    def __create_activity_edges(self):
         """
-        Iteratively run through self.simulation_graph and find all critical edges
+        Create nodes and edges for each individual activity.
+
+        Converts all cp_activity_ids within ``self.recorded_activity_df`` into start and end nodes
+        connected through an edge. All relevant attributes, such as the
+        duration of an activity, are added to these nodes and edges.
         """
-        pass
+        cp_activities_df = self.recorded_activity_df.drop_duplicates(
```

Comment:
is dropping of duplicates still needed here? It contains no duplicates by construction, right?

Reply:
false, we drop duplicates on subset 'cp_activity_id'. Duplicates may occur based on this subset since the (same) (loading) activity may be listed for both a vessel and a site.

```python
  cp_activities_df = self.recorded_activity_df.drop_duplicates(
            subset=["cp_activity_id"]
        )
```